packages/compiler-cli/src/ngtsc/extended_template_diagnostics/api/api.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {TmplAstNode} from '@angular/compiler';
+import {ParseSourceSpan} from '@angular/compiler/src/parse_util';
+import * as ts from 'typescript';
+
+import {ErrorCode} from '../../diagnostics';
+import {TemplateTypeChecker} from '../../typecheck/api';
+
+/**
+ * A Template Check receives information about the template it's checking and returns
+ * information about the diagnostics to be generated.
+ */
+export interface TemplateCheck {
+  /** Unique template check code, used for configuration and searching the error */
```

Comment:
```suggestion
  /** Unique template check code, used for configuration and searching the error. */
```
super nit: end the documentation sentences with a period (here everywhere else)